kobo/apps/openrosa/apps/api/tests/viewsets/test_xform_submission_api.py

```diff
@@ -438,3 +455,105 @@ def test_submission_blocking_flag(self):
                 )
                 response = self.view(request, username=username)
                 self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class ConcurrentSubmissionTestCase(RequestMixin, LiveServerTestCase):
```

Comment:
As @jnm's asked in former PR [[1](https://github.com/kobotoolbox/kobocat/pull/876#discussion_r1253934049)], ][2](https://github.com/kobotoolbox/kobocat/pull/876#discussion_r1210308268)], do we have tests that cover 

- client uploads XML with no attachment -> accept
- client uploads same XML again -> reject
- client uploads same XML but now with an attached file -> accept; attach file to existing submission

Reply:
Yes we have in `TestFormSubmission`

Reply:
For the records, the tests below are covering different scenarios:

1) `kobo/apps/openrosa/apps/api/tests/viewsets/test_xform_submission_api.py::ConcurrentSubmissionTestCase::test_post_concurrent_same_submissions` blocks submissions with same data `POST`ed concurrently

2) `test_duplicate_submission_with_different_content` and `test_duplicate_submission_with_same_content_but_with_attachment`